sqrl-server/sqrl-server-vertx/src/main/java/com/datasqrl/graphql/postgres_log/PostgresListenNotifyConsumer.java

```diff
@@ -65,16 +65,22 @@ public void handleNotification(PgNotification notification, Consumer<Object> lis
 
     List<Object> paramObj = new ArrayList<>();
 
-    // TODO (Soma) - We need to properly handle the payload types here.
-    //  This is hard coded currently and tailored to the sensors example.
+    // TODO (Soma) - This could be further improved. We chose json because that keeps the
+    //  datatype information, but Dates are an exception. Maybe we need to save the datatype
+    //  to the plan.
     for (String parameter : parameters) {
-      if (parameter.equals("timeSec")) {
-        paramObj.add(OffsetDateTime.parse(jsonPayload.getString(parameter)));
+      Object value = jsonPayload.getValue(parameter);
 
-      } else if (parameter.equals("sensorid")) {
-        paramObj.add(jsonPayload.getInteger(parameter));
+      if (value instanceof String) {
+        // Check if it's a datetime string and parse it
+        String strValue = (String) value;
+        if (strValue.matches("^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}.*$")) {
```

Comment:
Why does this require special handling for postgres @nandorsoma ? We are also consuming json data in the kafka listener what is the issue with reusing that code?